### PR TITLE
Readme collect & sort script

### PR DESCRIPTION
**Pull request by Jesse-NH**
Python script that collects all the readmes in NuBots folders and sorts them to be placed in the NuBook

- Filters through NuBots directory and pulls the README.md files
- Merges all of the Readmes from the same sub-folders into one large pages
- Sorts tthese into appropriate folders in the NuBook
- Names the files & fills out the header

### CODE REVIEW

File: tools/collect-readme.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+                folder4 = ''
+            else:
+                folder4 = folder4+' '
+            markdown_content = md_content_first+parent_folder.title()+md_content_second+list(slug_obj.parts)[-3].title()+md_content_third+list(slug_obj.parts)[-3]+' '+folder4+md_content_fourth+slug+md_content_fifth
+            file_path = Path(file)
+            item_path2 = list((Path(file)).parts)
+            #print(list(file_path.parts))
+
+            name_no_md = list(file_path.parts)[5].removesuffix('.md')
+            with open(file,'r') as firstfile, open('/home/nubots/NUbook/src/book/04-readmes/'+list(slug_obj.parts)[4]+'/'+name_no_md+'.mdx','a+') as secondfile:
```
Comment (Tom0Brien) [inline tools/collect-readme.py:115]: Make NUBook path an input relative to this script location

File: tools/collect-readme.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+                '''
+                # read content from first file
+                for lines in firstfile:
+                    # append content to second file
+                    secondfile.write(lines)'''
+                firstfile.close()
+                secondfile.close()
+    return
+
+list_of_dirs = ['/home/nubots/NUbots/module', '/home/nubots/NUbots/nuclear', '/home/nubots/NUbots/nusight2', '/home/nubots/NUbots/tools', '/home/nubots/NUbots/roles']
```
Comment (Tom0Brien) [inline tools/collect-readme.py:146]: Make these paths relative the script location

File: tools/collect-readme.py [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+            #print(list((Path(item)).parts))
+            try:
+                pass
+                #os.makedirs('/home/nubots/NUbook/src/book/04-readmes/'+folder_num+'-'+main_path+'/'+item_path+'/')
+            except: pass
+            for item2 in glob.glob(item+'*/'):
+                try:
+                    item_path = list((Path(item)).parts)[-1]
+                    main_path = list((Path(item)).parts)[-2]
+                    #os.makedirs('/home/nubots/NUbook/src/book/04-readmes/'+folder_num+'-'+main_path+'/'+item_path+'/')
```
Comment (Tom0Brien) [inline tools/collect-readme.py]: Remove commented out code

**Comment by Tom0Brien** [comment]
Run  `./b format` in a terminal inside the `NUbots` directory, it should point out some style changes to fix up

File: tools/collect-readme.py (hunk 2 above)
Comment (JosephusPaye) [reply tools/collect-readme.py]: I'd say take all paths as arguments the script: path to NUbots root (from which you can derive the sub paths to modules, NUsight, etc) and path to NUbook for output.

Long term though I think this is something that should live in NUbook, which can be triggered by a merge to NUbots main. That will then pull the latest NUbots changes and build it into NUbook, without committing any of the intermediate markdown files. All done in NUbook CI, with the only external input being a merge to NUbots main that triggers the NUbook build.

**Pull request closed without merging**